### PR TITLE
class methoda instance methoda and constructor

### PR DESCRIPTION
**Pull request by nawal-neeraj**
**class files**

- base_class_clsmethod.py
- base_class_const.py
- base_class_method.py

**method calling files:**

- class_method_update.py
- instance_construct.py
- instance_method.py

### CODE REVIEW

**Comment by rakeshcusat** [comment]
@nawal-neeraj please avoid checkin `*.pyc` files those are binary files and should not be checked in. you can add. them in the ignore file.

also, I am not sure why did you remove the old files. You can have kept them as it is.

File: method_example/base_class_clsmethod.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        self.instance_varable = val
+        
+    def show_variable(self):
+        print(f"instance variable update direct from init: {self.instance_varable}")
+        
+    def show_class_variable(cls):
+        print(f"class variable update direct from init: {cls._classvariable}")
+        
+    def update_instance_variable(self):
+        self.instance_varable = None
```
Comment (rakeshcusat) [inline method_example/base_class_clsmethod.py]: this is not a meaningful operation, modify this code to make it meaningful. You will not modify the variable with None in most of the cases right?

File: method_example/base_class_const.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        self.instance_varable = val
+        
+    def show_variable(self):
+        print(f"instance variable update direct from init: {self.instance_varable}")
+        
+    def show_class_variable(cls):
+        print(f"class variable update using class method: {cls._classvariable}")
+        
+    def update_instance_variable(self):
+        self.instance_varable = None
```
Comment (rakeshcusat) [inline method_example/base_class_const.py]: same comment as above

File: method_example/base_class_method.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class BaseClass:
+    
+    _classvariable = None
+    
+    def __init__(self):
+        self.instance_varable = None
+        
+    def show_variable(self,):
```
Comment (rakeshcusat) [inline method_example/base_class_method.py]: you don't need comma after `self`

File: method_example/base_class_const.py (hunk 2 above)
Comment (nawal-neeraj) [reply method_example/base_class_const.py]: Actually these methods have not called thats why I set the value of None. 
In https://github.com/nawal-neeraj/pytest/blob/feature-methods/method_example/base_class_method.py#L15 this line when I am using method call then I used parameter to update instance variable

File: method_example/base_class_clsmethod.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    def __init__(self, val):
+        self.instance_varable = val
+        
+    def show_variable(self):
+        print(f"instance variable update direct from init: {self.instance_varable}")
+        
+    def show_class_variable(cls):
+        print(f"class variable update direct from init: {cls._classvariable}")
+        
+    def update_instance_variable(self):
```
Comment (rakeshcusat) [inline method_example/base_class_clsmethod.py:14]:
```suggestion
    def update_instance_variable(self, val):
        self.instance_variable = val
```

File: method_example/base_class_const.py (hunk 2 above)
Comment (rakeshcusat) [inline method_example/base_class_const.py:15]: check my above comment.

Comment (rakeshcusat) [reply method_example/base_class_const.py]: The point is the method implementation is not making sense. You should review it by yourself and not have such code.

Comment (rakeshcusat) [reply method_example/base_class_const.py]: Other option is to make it abstract method.

**Review by rakeshcusat — APPROVED** [review]

**Pull request merged**